drivers/riello_ser.c

```diff
@@ -102,7 +102,6 @@ static int char_read (char *bytes, int size, int read_timeout)
 			return -1;
 		}
 		else {
-			bytes += now;
```

Comment:
I think this got inherited from some loop where the `bytes` pointer got incremented so we read next portions into the buffer while we can. The `read(..., size - readen)` above points to this too - with the current code, `readen` is always zero until we increment it with `now`.

Probably the current routine can be simplified even further to drop `readen` altogether, `return now` here and `return 0` in default exit below. But I do not know the legacy context behind it... @aquette @zykh @clepple : any memories on this one?

Reply:
Indeed this looks leftover from some old refactoring.  I'm inclined not to attempt any simplifications here.